File: array-splitting/array-splitting.py
def solution(A):
    if len(A) < 2:
        return 0
    split_index = -1
    score = 0
    left = 0

    right = sum(A)
    for i in range(len(A)):
        left += A[i]
        right -= A[i]

        if left == right:
            split_index = i + 1
            score += 1
            break

    if split_index != -1:
        score += max(solution(A[:split_index]), solution(A[split_index:]))
    return score

T = int(input())

for t in range(T):
    N = int(input())
    A = list(map(int, input().split(" ")))
    print(solution(A))

File: array-splitting/array-splitting.swift
import Foundation

func solution(_ A: [Int]) -> Int {
    if A.count < 2 {
        return 0
    }

    var splitIndex = -1
    var score = 0
    var left = 0
    var right = A.reduce(0, +)

    for i in 0..<A.count {
        left += A[i]
        right -= A[i]

        if left == right {
            splitIndex = i + 1
            score += 1
            break
        }
    }

    if splitIndex != -1 {
        score += max(solution(A[0..<splitIndex].map { $0 }), solution(A[splitIndex..<A.count].map { $0 }))
    }
    return score

}

let T = Int(readLine()!)!

for i in 0..<T {
    let N = Int(readLine()!)!
    let A = readLine()!.components(separatedBy: " ").map { Int($0)! }
    print(solution(A))
}
